slm_porosity/heat_transfer.f90

```diff
@@ -82,13 +83,16 @@ END FUNCTION func_with_derivative
   REAL(pr) :: fusion_heat
 
   ! bed parameters
-  REAL(pr) :: laser_power
-  REAL(pr) :: scanning_speed
   REAL(pr) :: initial_porosity
   REAL(pr) :: convective_transfer
   REAL(pr) :: radiative_transfer
   REAL(pr) :: absolute_temperature
 
+  ! laser parameters
+  REAL(pr) :: laser_power
+  REAL(pr) :: scanning_speed
+  REAL(pr) :: laser_beam_depth
```

Comment:
Laser absorption depth is about 100 times lesser than radius of the laser beam (100nm - 1mkm). So the absorption depth term is irrelevant. Precise name would be something like: halfwidth of laser gauss distribution in depth direction. Nevertheless, it is also could be wrong if we relabel FWHM to 3sigma. This value is characteristic of laser beam dimensions. There are depth in dimensions, but no absorption.

Reply:
Why does it physically correct to increase the absorption depth into 100 times?

Reply:
It is numerical approximation. With normalisation amount of the heat is the same.

Reply:
Why does this numerical approximation is physically correct? It can dramatically change the melting pool shape. For example, smoothing of the interface do not change it up to some extend (and I have checked it), but can possibly impact on grain structure. So I need justification of your suggestion from you. My offer: introduce the real `absorption_depth` and numerical `absorption_depth_factor`. In this case we can feel, what we do. Next, we need to simulate various values of `absorption_depth_factor`. After analysis, we can conclude, for example, that `absorption_depth_factor<50` does not violate the resulting temperature distribution. Finally, this story is not about laser alone.

Reply:
Which range of values for `depth` did your mean?